src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -238,6 +238,12 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         public static bool TryCoerceToRecord(this RecordValue value, RecordType targetType, out RecordValue result)
         {
             result = null;
+
+            if (!CheckRecordTargetMatching(value.Type, targetType))
+            {
+                return false;
+            }
+
             int n = value.Fields.Count();

```

Comment:
Do we want to support this scenario? I think this is a common scenario when value has more fields then targetType.
Now if we don't change the loop after this line, only an exact match will work and I don't think it is expected
```
var record = FormulaValue.NewRecordFromFields(new NamedValue("key", FormulaValue.New(1)), new NamedValue("value", FormulaValue.New(2)));
var recordType = RecordType.Empty().Add(new NamedFormulaType("key", FormulaType.Number));
record.TryCoerceTo(recordType, out var newResult);
```

Reply:
Updated code. we will fill blank for extra/missing field